src/k8s-engine/kim.ts

```diff
@@ -0,0 +1,150 @@
+const { EventEmitter } = require('events');
+const process = require('process');
+const { spawn } = require('child_process');
+const os = require('os');
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+const paths = require('xdg-app-paths')({ name: 'rancher-desktop' });
+const resources = require('../resources');
+
+const REFRESH_INTERVAL = 5 * 1000;
+
+interface childResultType {
+  stdout: string,
+  stderr: string,
+  code: number
+}
+
+interface imageType {
+  imageName: string,
+  tag: string,
+  imageID: string,
+  size: string,
+}
+
+export default class Kim extends EventEmitter {
+  constructor() {
+    super();
+    this.showedStderr = false;
+    this.refreshInterval = null;
+    this.currentCommand = null;
```

Comment:
Since these are static values, it might be better to declare them as properties (with comments):
```typescript
export default class Kim extends EventEmitter {
  /** Whether we have already displayed stderr for… */
  protected showedStderr = false;
  /** The timer for refreshing images */
  protected refreshInterval: ReturnType<typeof setInterval> | null = null;
  /** The most recently run command */
  protected currentCommand: string | null = null;
```
(`refreshInterval` can have a clearer type if we import the `timers` module, I think?)

Reply:
Not sure about timers -- it hasn't been touched in 7 years (not that there's much in it to go wrong) and github doesn't recognize the MIT License because of the typo. Seems too leftpadish to get a better type.

Reply:
Not the NPM package, but the Node [builtin](https://nodejs.org/docs/latest-v14.x/api/timers.html).